### ISSUE TITLE
TS classes have automatic constructors

### ISSUE THREAD

**Issue by ctaggart** [fable-compiler/ts2fable#83]
8.3.3 Automatic Constructors
https://github.com/Microsoft/TypeScript/blob/master/doc/spec.md#8.3.3

While looking into gRPC https://github.com/fable-compiler/Fable/issues/1183, I had to add the default constructor manually. It should be generated.

```
ts2fable ..\..\ts\_proto\examplecom\library\book_service_pb.d.ts book_service_pb.fs
```

It should have:
``` fs
type [<AllowNullLiteral>] GetBookRequestStatic =
    [<Emit "new $0($1...)">] abstract Create: unit -> GetBookRequest
```

**Comment by ctaggart** [comment]
Mocha is a good test case. A new complication is that the constructor is inherited if not defined. If defined in the base class, those constructors are inherited.

![image](https://user-images.githubusercontent.com/80104/32867055-9cfb55f6-ca49-11e7-9e31-9ce31817086f.png)

**Comment by ctaggart** [comment]
I haven't been able to figure out how to look up the base type given the `extends` reference. I opened up:
https://stackoverflow.com/questions/47340859/how-to-i-get-the-declared-type-from-a-typereferencetype-in-typescript-compiler-a

**Comment by ctaggart** [comment]
I create a test case in TypeScript and added https://github.com/Microsoft/TypeScript/issues/20098 as well.

**Comment by ctaggart** [comment]
I was not able to figure out how to navigate using the TypeScript TypeChecker, so I used another solution. I was able to figure out how to get the full name for the references and the classes, so I used that information to copy the constructors.

**Comment by ctaggart** [comment]
fixed by #86
